In [1]:
#DEPENDENCIES & INITIALISATIONS
from IPython.display import HTML #For embedding videos
import math #Basic math functionality
import numpy as np #Numpy for working with arrays
import matplotlib.pyplot as plt #Plotting functionality 
#HTML('<center><iframe width="560" height="315" src="#" frameborder="0" allowfullscreen></iframe></center>')
print("- Dependencies imported 👍")
print("- Remember to select 'Kernel-> Restart & Run All' on first load to show embedded Youtube videos 🎬")
print("- Author: Dr SPCarroll, Sept 2020, Rev 0")

- Dependencies imported 👍
- Remember to select 'Kernel-> Restart & Run All' on first load to show embedded Youtube videos 🎬
- Author: Dr SPCarroll, Sept 2020, Rev 0


# Week 1: Pressure Vessels
Reference: Chapter 8 of *Mechanics of Materials*, Gere

![title](img/T2W1/banner.jpg)

## 1.0 Introduction

Continuing the theme of hydrostatics and the influence of pressure, this week we will consider the behaviour of *pressure vessels*. Pressure vessels are commonly found in everyday life, from soft drink cans to scuba tanks to large scale industrial containers processing plants. These are closed structures that contain liquid or gas under pressure. For the purpose of our discussion here we will assume a positive gauge pressure within the vessels under consideration. Our task is to understand the stresses that develop within the walls of these structures. 

We will find that a particular state of stress develops in the walls of these structures called **plane stress**. We will briefly introduce and demonstrate this concept here but you will study plane stress extensively in your Solid Mechanics module next year (see chapter 7 of *Mechanics of Materials* by Gere for a head start). 

All of the pressure vessels we consider here are **thin-walled**. This means their wall thickness, $t$ is small compared to their radius, $r$. We will loosely classify a pressure vessel as thin-walled if it has a ratio of radius to wall thickness, 
\begin{equation}
\frac{r}{t}\geq 10
\end{equation}

If this condition is not satisfied, more sophisticated methods, beyond simple statics, are needed to model the stresses developed within the structure. 

## 2.0 Spherical pressure vessels

We begin by considering a spherical pressure vessel, similar to that shown below. 

![title](img/T2W1/img2.jpg)
<center>Spherical pressure vessel <a href="https://commons.wikimedia.org/wiki/File:BS_Kugelgasbehaelter.JPG">[Wikipedia]</a></center>

We will neglect the reaction forces imposed by the supports and any other externally applied forces. For now we simply consider a spherical tank with an internal gauge pressure $p$. Now imagine that we slice vertically through the sphere, dividing it equally into 2 hemi-spheres. The image below shows the cross-section, with internal pressure $p$ pressing against the walls of the sphere. 

![title](img/T2W1/img3.jpg)
<center>Cross-section of the sphere indicating the internal pressure $p$.</center>

We can now consider force equilibrium of the hemisphere, shown below. The internal pressure acts against the curved internal surface of the hemisphere. We know from our study of hydrostatics that the resultants force can be obtained by applying the pressure $p$ onto the vertical projection of the curved surface, which is the circle with radius $r$. Therefore the pressure resultant is obtained as,

\begin{equation}
P = p\times \pi r^2
\end{equation}

In response to the gauge pressure, tensile stresses develop in the walls of the tank. In making a vertical cut through the tank, we have revealed these internal tensile stresses, $\sigma$. If we define $\bar{r}$ as the mean radius, 

\begin{equation}
\bar{r} = r + \frac{t}{2}
\end{equation}

the resultant tensile force in the sphere wall is,

\begin{equation}
F_t = \sigma\times 2\pi \bar{r}\times t
\end{equation}

Now equation the forces $P$ and $F_t$ and solving for $sigma$ yields,

\begin{align}
p\times \pi r^2 &= \sigma\times 2\pi \bar{r}\times t\\
\sigma &= \frac{pr^2}{2\bar{r}t} 
\end{align}

Since we have made the initial assumption that $t<<r$, we can simplify our expression by replacing $\bar{r}$ with the internal radius $r$. Although either value of radius will give an acceptable answer, using the internal radius yields an answer closer to the true value derived from a more in-depth analysis. Therefore we can state our approximation for the tensile stress in a thin walled spherical pressure vessel as,

\begin{equation}
\boxed{
\sigma=\frac{pr}{2t}
}
\end{equation}

![title](img/T2W1/img4.jpg)
<center>Hemisphere and liquid isolated by the cut.</center>

It should be apparent that once we slice through the centre of the sphere, the orientation of the slice will not change the result. As such we can conclude that a uniform tensile stress $\sigma$, develops in the walls of the spherical pressure vessel. We can visualise this by sketching mutually perpendicular (orthogonal) tensile stresses on a small *stressed element* on the surface of the sphere.

![title](img/T2W1/img5.jpg)
<center>Tensile stresses developed in the wall of spherical pressure vessel.</center>

### 2.1 Stresses at the outer surface

We now consider a small stress element extracted from the outer surface of the sphere. We observe mutually perpendicular stresses, labeled in the figure below as $\sigma_x$ and $\sigma_y$. In this element, the $x$ and $y$ axes indicated are tangential to the surface of the sphere, while the z-axis is perpendicular to the surface. On the outer surface of the pressure vessel we note that there is no external forces acting and as such there is no stress developed on the z-direction. The only stresses developed on the outer surface are tensile stresses $\sigma_x$ and $\sigma_y$. We can say that this stress element and by extension, the outer surface is in a state of **biaxial stress**.   

![title](img/T2W1/img6.jpg)
<center>Stresses on the outer surface of the sphere.</center>

### 2.2 Stresses at the inner surface

If we now extract a cuboid stress element from the inner surface we see that we now have a pressure $p$ acting in the z-direction and as such an element at this location is in a state of triaxial stress. The stress in the z-direction, perpendicular to the surface, decreases from a value of $p$ at the inner surface to $0$ at the outer surface. 

Remember that our analysis has made the assumption that we can neglect the wall thickness, $t$. As such our model is an approximation of the actual behaviour of the structure. Consistent with this approximate approach, if the structure is thin-walled, the magnitude of $\sigma_z$ will be small by comparison to $\sigma_x$ and $\sigma_y$. We therefore make the simplifying assumption that the inner surface is also in a state of biaxial stress. 

![title](img/T2W1/img7.jpg)
<center>Stresses on the inneer surface of the sphere.</center>

**Observations and limitations**
- Our analysis has assumed a perfectly spherical vessel free from any point loads or discontinuities in its geometry. In reality, such a container will have supports, fittings etc. These will introduce **stress concentrations**. These are localised regions of higher magnitude stress. You should be aware that the formula derived above will only apply in regions free from stress concentrations. Other method of analysis, typically finite element analysis, are needed to model the influence of stress concentrations. 
- Remember the underlying assumption we made was that the wall thickness is small compared to the radius. This allows us to neglect the stress $(\sigma_z)$ gradient between the inner and outer walls and assume the walls are in a state of biaxial or plane stress. 
- We have assumed a positive gauge pressure, i.e. the internal pressure acts like air in a balloon, pushing the walls outward. Insufficient gauge pressure could lead to inward buckling of the walls of the shell. 
- All of the stresses we have discussed are perpendicular to the faces of the stress element and are therefore known as **normal stresses**. Shear stresses *would* act perpendicular to the faces of the stress element. For now we will simply note that there are no in-plane shear stresses acting on the faces of the element. In your further study of stress analysis, you will see that $\sigma_x$ and $\sigma_y$ are in fact principal stresses and as such there can be no shear stresses acting on the same planes as $\sigma_x$ and $\sigma_y$. For now it is sufficient to concentrate on normal stresses identified above. 

## 3.0 Cylindrical pressure vessels

Perhaps even more common than the spherical pressure vessel is the cylindrical pressure vessel, pictures below. In this section we will again use simple statics to determine the stresses that develop in the walls of these structures. The same assumptions discussed above regarding thin walls and stress concentrations also apply to cylindrical pressure vessels. 

![title](img/T2W1/img8.jpg)
<center>Cylindrical pressure vessel. <a href="https://en.wikipedia.org/wiki/Storage_tank#/media/File:Modified_Hanson_steelwatertank.jpg">[Wikipedia]</a></center>

Let us consider a cylindrical pressure vessel free from any stress concentrations. A stress element on the surface of the tank with sides parallel and perpendicular to the logitudinal axis of the tank would have stresses $\sigma_1$ and $\sigma_2$ as shown below. These are principal stresses and no shear stresses act on these faces (you'll cover this in more depth in the future). However, unlike the spherical pressure vessel, $\sigma_1 \neq \sigma_2$; $\sigma_1$ is referred to and the **circumferential or hoop stress** while $\sigma_2$ is referred to as the **longitudinal or axial stress**. 

![title](img/T2W1/img9.jpg)
<center>Cylindrical pressure vessel with hoop and axial stress indicates as $\sigma_1$ and $\sigma_2$ respectively.</center>

### 3.1 Hoop Stress

Consider making cuts and isolating a segment of the pressure vessel (and its enclosed liquid) pictured below. In doing so, we reveal the internal hoop stresses. Just as we did for the spherical pressure vessel above, we can equate the tensile force generated by the hoop stress $F_{\text{hoop}}$ with the force resultant $P_1$ from the internal liquid pressure $p$,

\begin{align}
F_{\text{hoop}} &= P_1\\
\sigma_1\times 2bt &= p\times 2rb
\end{align}

From this we can extract an expression for the hoop stress in a thin-walled cylindrical pressure vessel,

\begin{equation}
\boxed{
\sigma_1 = \frac{pr}{t}
}
\end{equation}

![title](img/T2W1/img10.jpg)
<center>Segment and liquid/gas isolated by the cut revealing internal hoop stresses, $\sigma_1$</center>

### 3.2 Axial Stress

To determine an expression for the axial stress we will again isolate a portion of the vessel and its liquid. Considering force equilibrium now in the axial direction,

\begin{align}
F_{\text{axial}} &= P_2\\
\sigma_2\times 2\pi r t &= p\times \pi r^2
\end{align}

This yields an expression for the axial stress,

\begin{equation}
\boxed{
\sigma_2 = \frac{pr}{2t}
}
\end{equation}

We note that the axial stress is equal to the tensile stress in a spherical pressure vessel. We also note that the hoop stress is two times larger than the axial stress for a cylindrical pressure vessel,

\begin{equation}
\boxed{
\sigma_1 = 2\sigma_2
}
\end{equation}


![title](img/T2W1/img11.jpg)
<center>Segment and liquid/gas isolated by the cut revealing internal axial stresses, $\sigma_2$</center>

### 3.3 Stresses on the inner and outer faces

We can again compare the stresses that develop on the inner and outer faces of the pressure vessel. The same results apply that we observed for spherical pressure vessels in sections 2.1 and 2.2 above. The stress, $\sigma_z = \sigma_3$ normal to the surface of the wall, varies from $\sigma_3 = -p$ at the inner surface to $\sigma_3 = 0$ on the outer surface. However, in line with the thin-walled assumption, we can neglect this stress as it is small in comparison to the membrane stresses $\sigma_1$ and $\sigma_2$. The equations derived for hoop stress and axial stress are as mentioned above only valid in regions without stress concentrations. They are also limited to regions where there are no discontinuities in the geometry. Therefore they will not apply towards the ends of the cylinder near the end caps. More sophisticated methods of analysis are required to determine the stresses in these regions. 

### Example 1
(Ref. Gere)

A steel compressed air tank is formed by welding 2 hemispheres together and has inner diameter of $450\:mm$ and a wall thickness of $7 mm$. 
- If the allowable tensile stress in the steel is $115\: N/mm^2$, what is the maximum allowable pressure?
- Tests on the weld show that it will fail if the load on the weld exceeds $1.5\:MN/m$ of weld. If the required factor of safety against failure of the weld is 2.5, what is the maximum allowable pressure? You should assume the weld has the same depth as the vessel wall, i.e. it is a *full penetration butt weld*.

In [9]:
#Steel Failure - Rearrange equation for tensile stress in spherical pressure vessel
r = 450/2 #(mm) Vessel radius
t = 7 #(mm) Wall thickness
sigma_all = 115 #(N/mm^2) Allowable tensile stress in the steel
p_all_1 = 2*t*sigma_all/r #(N/mm^2) Allowable internal pressure
print("- The allowable internal pressure based on steel failure is {one} N/mm^2".format(one=round(p_all_1,2)))

- The allowable internal pressure based on steel failure is 7.16 N/mm^2


In [12]:
#Weld Failure
T_all = 1.5/2.5*10**3 #(N/mm) Allowable load on the weld per mm
sigma_all = (T_all*1)/(t*1) #(N/mm^2) Allowable stress in the self (force/area)
p_all_2 = 2*t*sigma_all/r #(N/mm^2) Allowable internal pressure
print("- The allowable internal pressure based on weld failure is {one} N/mm^2".format(one=round(p_all_2,2)))


- The allowable internal pressure based on weld failure is 5.33 N/mm^2


Therefore the maximum internal pressure in the vessel is $5.33\:N/mm^2$. Note that this analysis has not taken into consideration any shear stresses that develop within the walls of the vessel on planes other than the principle planes (upon which the principal tensile stresses develop).

**You should now download the question sheet for this week and attempt all questions. You will have an opportunity to to get help in your tutorial session and worked solutions will be released next week.**  